aloha-core/src/main/scala/com/eharmony/aloha/audit/impl/scoreproto/ScoreAuditor.scala

```diff
@@ -0,0 +1,254 @@
+  implicit def doubleAuditor: ScoreAuditor[Double] = DoubleAuditor
+  implicit def stringAuditor: ScoreAuditor[String] = StringAuditor
+
+  /**
+    * Cast `auditor` to `ScoreAuditor[N]` and wrap in an Option.
+    * '''NOTE''': ''I hate this method'' but I don't want to require an implicit
+    * [[ScoreAuditor]] as a parameter in the apply method because [[ScoreAuditorImpl]]'s
+    * `changeType` method would then need the same implicit to be included.  A determination
+    * should be made as to the feasibility of this strategy.
+    * @param auditor an auditor to wrap
+    * @tparam N type of auditor to be returned
+    * @return
+    */
+  private[this] def opt[N](auditor: ScoreAuditor[_]): Option[ScoreAuditor[N]] =
+    Option(auditor.asInstanceOf[ScoreAuditor[N]])
+
+  private[this] sealed trait ScoreAuditorImpl[N] extends ScoreAuditor[N] {
+
+    /**
+      * Convert an audited value of the natural type to a `BaseScore.Builder`.
+      * This builder will be augmented with the model ID and possibly a probablity of
+      * the score being produced randomly.  This builder will be built prior to return
+      * from the [[MorphableAuditor]] API.
+      *
+      * @param valueToAudit a value that is audited
+      * @return a BaseScore.Builder based on the value being audited.
+      */
+    def boxScore(valueToAudit: N): BaseScore.Builder
+
+    override final def changeType[M: RefInfo]: Option[ScoreAuditor[M]] = ScoreAuditor[M]
+
+    override final def failure(key: ModelIdentity,
+                               errorMsgs: => Seq[String],
+                               missingVarNames: => Set[String],
+                               subValues: Seq[Score]): Score = {
+      val b = Score.newBuilder
+      if (subValues.nonEmpty)
+        b.addAllSubScores(asJavaIterable(subValues))
+      val e = ScoreError.newBuilder.setModel(mId(key)).addAllMessages(asJavaIterable(errorMsgs))
+      val m = accumulateMissing(missingVarNames, subValues)
+      if (m.nonEmpty)
+        e.setMissingFeatures(MissingRequiredFields.newBuilder.addAllNames(asJavaIterable(m)))
+      b.setError(e).build
+    }
+
+    override final def success(key: ModelIdentity,
+                               valueToAudit: N,
+                               errorMsgs: => Seq[String] = Nil,
+                               missingVarNames: => Set[String] = Set.empty,
+                               subValues: Seq[Score] = Nil,
+                               prob: => Option[Float] = None): Score = {
+      val m = mId(key)
+      val b = Score.newBuilder
+      if (subValues.nonEmpty)
+        b.addAllSubScores(asJavaIterable(subValues))
+      val mm = accumulateMissing(missingVarNames, subValues)
+      val e = errorMsgs
+      if (mm.nonEmpty || e.nonEmpty) {
+        val eb = ScoreError.newBuilder.
+          setMissingFeatures(MissingRequiredFields.newBuilder.addAllNames(asJavaIterable(mm))).
+          setModel(m)
```

Comment:
Why is the model id and name only set in the error flow?  Does this pertain to the submodel identity during a recursive tree traversal?